device_backends/LogicalNameMapping/src/LNMMathPlugin.cc

```diff
@@ -219,6 +230,20 @@ namespace ChimeraTK { namespace LNMBackend {
 
   /********************************************************************************************************************/
 
+  bool MathPlugin::checkAllParametersWritten(
+      std::map<std::string, boost::shared_ptr<NDRegisterAccessor<double>>> const& accessorsMap) {
+    bool allUpdated = true;
+    auto backend = _backend.lock(); // get a shared pointer from the weak pointer
```

Comment:
Please add a comment why no check is necessary that the lock is successful (you may drop the comment what this line does, as it is very obvious).